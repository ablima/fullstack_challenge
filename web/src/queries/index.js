import gql from "graphql-tag";

export const GET_PRODUCTS = gql`
  query Products($id: Int, $name: String, $categoryId: Int){
    getProducts(id: $id, name: $name, categoryId: $categoryId){
      id
      name
      description
      qnt
      price
      thumbnails
    }
  }
`

export const ADD_PRODUCT_TO_CART = gql`
  mutation Product($id: Int){
    addProductToCart(id: $id){
      id
    }
  }
`;

export const REMOVE_PRODUCT_FROM_CART = gql`
  mutation Product($id: Int){
    removeProductFromCart(id: $id){
      id
    }
  }
`;

export const GET_CATEGORIES = gql`
  query {
    getCategories{
      id
      name
    }
  }
`;

export const CREATE_ORDER = gql`  
  mutation Order($order: NewOrder){
    createOrder(order: $order){
      id
    }
  }
`;