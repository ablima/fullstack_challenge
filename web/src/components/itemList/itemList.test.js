import React from 'react';
import renderer from 'react-test-renderer';
import { MockedProvider } from '@apollo/client/testing';
import { ADD_PRODUCT_TO_CART, REMOVE_PRODUCT_FROM_CART } from '../../queries';
import ItemList from './index';

const mocks = [
  {
    request: {
      query: ADD_PRODUCT_TO_CART,
      variables: {
        id: 17,
        qnt: 1
      }
    },
    result: {}
  },
  {
    request: {
      query: REMOVE_PRODUCT_FROM_CART,
      variables: {
        id: 17,
        qnt: 1
      }
    },
    result: {}
  }
];
  
const product = {
  name: "Produto",
  description: {},
  qnt: 5,
  price: 100,
  categoryId: 1,
  thumbnails: "images/celular1-1.jpg"
}

test('Check if render is correct', () => {
  const component = renderer.create(
    <MockedProvider mocks={mocks} addTypename={false}>
      <ItemList data={product} />
    </MockedProvider>
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});