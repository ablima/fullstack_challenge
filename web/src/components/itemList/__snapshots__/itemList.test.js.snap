// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check if render is correct 1`] = `
<div
  className="itemContainer item"
>
  <div
    className="infoContainer"
  >
    <img
      src="http://localhost:3010/static/i"
    />
    <h4>
      Produto
    </h4>
  </div>
  <div
    className="qntContainer"
  >
    <div
      className="priceContainer"
    >
      <h5>
        Preço:
      </h5>
      <h5>
        R$ 100,00
      </h5>
    </div>
    <div
      className="qntInputContainer"
    >
      <div>
        <button
          onClick={[Function]}
        >
          -
        </button>
        <input
          onBlur={[Function]}
          onChange={[Function]}
          value={5}
        />
        <button
          onClick={[Function]}
        >
          +
        </button>
      </div>
      <span
        onClick={[Function]}
      >
        Remover
      </span>
    </div>
    <div
      className="priceContainer"
    >
      <h5>
        Subtotal:
      </h5>
      <h5>
        R$ 500,00
      </h5>
    </div>
  </div>
</div>
`;
